Update ft_definetrial.m

NirOfir opened the pull request:
Commit #1575 added the undocumented but important field "representation", so I added it to the doc, and made small changes to the rest of the help section.

NirOfir commented:
On second thought, would it be better to have the default automatic, where the output of the trialfun is returned as is?

robertoostenveld commented:
> On second thought, would it be better to have the default automatic, where the output of the trialfun is returned as is?

Should the trialfun be allowed to decide here? I think that would make sense. The default trialfuns that we now have all (except for ft_trialfun_imotions) return a numeric representation, but for a custom trial fun that returns a table it should not be converted back to numeric. 

We usually use `[]` to code for the defaults. Can you add something like this

```
if isempty(cfg.representation)  
  % keep the format as is
elseif strcmp(cfg.representation, 'numeric') && istable(trl)
  % convert
elseif strcmp(cfg.representation, 'table') && isnumeric(trl)
  % convert
end
```

and change the default at the top into 

```
cfg.representation  = ft_getopt(cfg, 'representation'); % empty, 'numeric' or 'table'
```

NirOfir commented:
Thanks! I changed the code a bit:
1. I removed the cases for empty tables/matrices checks, becuase those would return an error anyway on line 191.
2. I added  a check to make sure there are at least 3 columns in trl.
3. I edited the conversion numeric -> table to be more concise (I couldn't come up with any actual differences from the previous method).
4. I also changed the conversion from table to numeric, because as it was it would replace trl with a NX3 matrix, and then look for more cols in trl, which it already removed.

Waiting for your feedback :)

robertoostenveld commented:
looks good to me, let's merge and check that all test scripts run through...
